### PR TITLE
Type conversions between arkworks and blst

### PR DESCRIPTION
**Pull request by huitseeker**
This PR follows #96, which should be merged before it. 

The conversions will allow systematic conversions arkworks <-> blst to check operations match.

### CODE REVIEW

**Review by kchalkias — COMMENTED** [review]
Love the approach + round trip tests. Is there any special representation for edge cases? Ie zero elements that needs different handling?

**Comment by huitseeker** [comment]
> Is there any special representation for edge cases? Ie zero elements that needs different handling?

Yes, essentially the infinity element for G1 & G2. 

Where I was going with this PR, considering the arkworks types are more used for testing than anything else (and that the arkworks serialization types are "wrong" in the sense that they do not follow the zcash spec), was to land the non-degenerate cases, to have an e2e test for Groth16, and then extend those conversion functions with the infinity representations.

File: fastcrypto-zkp/src/conversions.rs [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+
+/// Affine point translations: those mostly allow us to receive the
+/// proof points, provided in affine form.
+pub fn bls_g1_affine_to_blst_g1_affine(pt: &BlsG1Affine) -> blst_p1_affine {
+    debug_assert_eq!(pt.uncompressed_size(), G1_UNCOMPRESSED_SIZE);
+    let tmp_p1 = blst_p1_affine {
+        x: bls_fq_to_blst_fp(pt.x),
+        y: bls_fq_to_blst_fp(pt.y),
+    };
+    // See https://github.com/arkworks-rs/curves/issues/14 for why the double serialize
```
Comment (kchalkias) [inline fastcrypto-zkp/src/conversions.rs:136]: +1

Comment (kchalkias) [reply fastcrypto-zkp/src/conversions.rs:136]: It would be great to have a a bulletpoint list of what we offer to the community after this epic. BLST backend, better timings, fixes / extra safety over existing applications (correct masks etc), better alignment with standards etc. This can be part of a readme subsection re importance and promising results of our implementation. It can be a follow up PR.

**Review by kchalkias — APPROVED** [review]
Fantastic!
